setup_apimanager.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+# Prereq:
+# jq, oc, openshift cluster, ssh keys setup for github, operator-sdk, go1.19
+
+# Usage: 
+# chmod 771 setup_monitoring.sh
+# ./setup_apimanager.sh
+
+# Install 3scale-operator via olm
+oc apply -f - <<EOF
+---
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: 3scale-operator
+  namespace: 3scale-test
+spec:
+  channel: threescale-2.14
+  installPlanApproval: Automatic
+  name: 3scale-operator
+  source: redhat-operators
+  sourceNamespace: openshift-marketplace
+  startingCSV: 3scale-operator.v0.11.10
+EOF
+oc project 3scale-test
+# create dummy S3 secret
+oc apply -f - <<EOF
+---
+kind: Secret
+apiVersion: v1
+metadata:
+  name: s3-credentials
+data:
+  AWS_ACCESS_KEY_ID: UkVQTEFDRV9NRQ==
+  AWS_BUCKET: UkVQTEFDRV9NRQ==
+  AWS_REGION: UkVQTEFDRV9NRQ==
+  AWS_SECRET_ACCESS_KEY: UkVQTEFDRV9NRQ==
+type: Opaque
+EOF
+# Create Apimanager CR with monitoring enabled and system-database set to postgresql
+DOMAIN=$(oc get routes console -n openshift-console -o json | jq -r '.status.ingress[0].routerCanonicalHostname' | sed 's/router-default.//')
+oc apply -f - <<EOF
+---
+apiVersion: apps.3scale.net/v1alpha1
+kind: APIManager
+metadata:
+  name: apimanager-sample
+spec:
+  system:
+    monitoring:
+        enabled : true
+    database:
+        postgresql: {}
+    fileStorage:
+      simpleStorageService:
+        configurationSecretRef:
+          name: s3-credentials
+  wildcardDomain: $DOMAIN
+EOF
+# Check the install has completed
+while true; do
+    # Get the status and type fields of the custom resource
+    STATUS=$(oc get APIManager apimanager-sample -o=jsonpath='{.status.conditions[*].status}')
+    TYPE=$(kubectl get APIManager apimanager-sample -o=jsonpath='{.status.conditions[*].type}')
+
+    # Check if the desired condition is met
+    if [[ "$STATUS" == "True" && "$TYPE" == "Available" ]]; then
+        echo "Status: $STATUS, Type: $TYPE"
+        echo "ApiManager installation finished"
+        break
+    else
+        echo "Status: $STATUS, Type: $TYPE"
+        echo "ApiManager installing. Retrying in 10 seconds."
+        sleep 10
```

Comment:
Should we put a timeout to eventually cause the script to fail if the installation never reports as `Available`?

Reply:
have been looking into this apimanager failure ,  Routes never came up even though all the deploymentsConfigs completed may be something wrong with one of the components. Will try a fresh install today .  We shouldn't need this while loop this should be enough to check it 
```bash
oc wait --for=condition=available apimanager/apimanager-sample --timeout=-1s
```
but easier to put a time out in with the loop. 
I have this check in another PR which checks the deployments part of the status for when all are ready and none are stopped or starting
```bash
function deployments(){
    if [[ -z "$1" ]]; then
        echo "$1 variable is empty"
        return 0 
    else
        echo "$1 variable is not empty"
        return 1
    fi
}
while true; do
    STARTING=$(oc get apimanager apimanager-sample -o jsonpath="{.status.deployments.starting}")
    STOPPED=$(oc get apimanager apimanager-sample -o jsonpath="{.status.deployments.stopped}")
    READY=$(oc get apimanager apimanager-sample -o jsonpath="{.status.deployments.ready}")
    if deployments $STARTING && deployments $STOPPED && !(deployments $READY) ; then
        echo "All deployment conditions are satisfied for apimanger. Proceeding with the script."
        break  # Exit the loop when all conditions are satisfied
    else
        echo "One or more conditions are not satisfied for apimanager. Retrying in 10 seconds."
        sleep 10  # Wait for 10 seconds before retrying
    fi
done```
but think we should look into the apimanger failure rather that work around it .